ample/util/options_processor.py

```diff
@@ -81,63 +102,69 @@ def process_options(optd):
-
-    ###############################################################################
-    #
     # FASTA processing
-    #
-    ###############################################################################
     # Check to see if mr_sequence was given and if not mr_sequence defaults to fasta
     if optd['mr_sequence'] != None:
         if not (os.path.exists(str(optd['mr_sequence']))):
             msg = 'Cannot find mr sequence file: {0}'.format(optd['mr_sequence'])
             exit_util.exit_error(msg)
     else:
         optd['mr_sequence'] = optd['fasta']
-
     # Process the fasta file and run all the checks on the sequence
     sequence_util.process_fasta(optd, canonicalise=True)
-
-    #
+    
     # Not sure if name actually required - see make_fragments.pl
-    #
     if optd['name'] and len(optd['name']) != 4:
         msg = '-name argument is the wrong length, use 4 chars eg ABCD'
         exit_util.exit_error(msg)
-
     # Underscore required by rosetta make_fragments.pl
     optd['name'] += '_'
-
-    ###############################################################################
-    #
-    # Contact file processing
-    #
-    ###############################################################################
-
-    #if False:
-    if optd['contact_file'] or optd['bbcontacts_file'] or not optd["no_contact_prediction"]:
-        contact_util.ContactUtil.check_options(optd)
-        optd['use_contacts'] = True
-    ###############################################################################
-    #
     # MTZ file processing
-    #
-    ###############################################################################
     try:
         mtz_util.processReflectionFile(optd)
     except Exception, e:
         msg = "Error processing reflection file: {0}".format(e)
         exit_util.exit_error(msg, sys.exc_info()[2])
+    # Contact file processing
+    if optd['contact_file'] or optd['bbcontacts_file'] or not optd["no_contact_prediction"]:
+        contact_util.ContactUtil.check_options(optd)
+        optd['use_contacts'] = True
+    process_modelling_options(optd)
+    # Misc options
+    if optd['missing_domain']:
+        logger.info('Processing missing domain\n')
+        if not os.path.exists(optd['domain_all_chains_pdb']):
+            msg = 'Cannot find file domain_all_chains_pdb: {0}'.format(optd['domain_all_chains_pdb'])
+            exit_util.exit_error(msg)
+    process_mr_options(optd)
+    process_benchmark_options(optd)
 
-    ###############################################################################
-    #
-    # Modelling and ensemble options
-    #
-    ###############################################################################
+    logger.info('Running on %d processors' % optd['nproc'])
+    # cluster queueing
+    if optd['submit_qtype']:
+        optd['submit_qtype'] = optd['submit_qtype'].upper()
+    if optd['submit_cluster'] and not optd['submit_qtype']:
```

Comment:
Could we not deprecate the `-submit_cluster` flag? Not necessarily remove it entirely but add a warning message to the help menu and make it do nothing ...

Reply:
I agree, but that's something for a separate pull request